pkg/validation/eval.go

```diff
@@ -193,14 +214,22 @@ func (ev *Evaluator) Evaluate(rules []Rule, vm *k6tv1.VirtualMachine) *Result {
 		ra, err := r.Specialize(vm, refVm)
 		if err != nil {
 			fmt.Fprintf(ev.Sink, "%s failed: cannot specialize: %v\n", r.Name, err)
-			result.Fail(r, err)
+			if r.JustWarning {
```

Comment:
Here we should fail too. We are trying to apply the rule to the given VM, and somehow this fails. It's still a syntax processing step.

In general, there are two steps:
1. trying to make sense of a given rule (e.g. well formed, syntactically valid, express a meaningful rule)
2. apply the rule to a given VM, see if passes or not.

The step #2 is the only step which needs to be skipped with a warning, we always want to check rules are correct and meaningful, don't we?
If we *do not*, however, we should address in a separate patch.